help/develop/javascript-development/packages.md

```diff
@@ -0,0 +1,263 @@
+register your viewers, widgets, renderers, converters, validators, suggestions, information panels, and semantic type 
+detectors.
+
+If a function has more than one output, it must return a JavaScript object `{prop1: value1, prop2: value2}`, for 
+example:
+
+```js
+//name: test1
+//output: string a1
+export function test1() {
+  return 'a';
+}
+//name: test2
+//output: string s1
+//output: string s1
+export function test2() {
+  return {s1: 'a', s2: 'b'};
+}
+```
+
+### <a href="#" id="detectors.js"></a>detectors.js
+
+The `detectors.js` file must define a `<packageName>PackageDetectors` class that extends the base `DG.Package` class.
+(For more information about this class, refer to the [JavaScript API - Package section].)
+`detectors.js` is similar to `package.js` but is intended for smaller functions called [_semantic type detectors_],
+which are loaded separately from the rest of the package. Semantic type detectors are called each time a user opens a 
+table in Datagrok and are used to quickly inspect the data and determine the semantic type of the columns.
+
+Tagging the semantic types allows Datagrok to offer specific functions for data of a particular type.
+
+Here's an example from the [Sequence package] that contains the `detectNucleotides` detector:
+
+```js
+class SequencePackageDetectors extends DG.Package {
+    
+    //tags: semTypeDetector
+    //input: column col
+    //output: string semType
+    detectNucleotides(col) {
+        if (col.name.startsWith('nuc')) {
+            col.semType = 'nucleotides';
+            return 'nucleotides';
+        }
+        return null;
+    }
+}
+```
+
+Once registered by Datagrok, this function becomes available across the whole platform and can be used for semantic type 
+detection.
+
+### <a href="#" id="webpack.config.js"></a>webpack.config.js
+
+The Grok CLI generates a typical webpack configuration for the Datagrok package:
+
+```javascript
+const path = require('path');
+
+module.exports = {
+    mode: 'development',  // Set to "production" to minify the output and optimize production builds
+    entry: {
+        package: './src/package.js'  // The package is limited to exactly one entry point
+    },
+    devtool: 'inline-source-map',   // Enhances package debugging in browser devtools
+    externals: {                    // The external modules won't be loaded to the output, but taken from the environment
+        'datagrok-api/dg': 'DG',
+        'datagrok-api/grok': 'grok',
+        'datagrok-api/ui': 'ui',
+        'openchemlib/full.js': 'OCL',
+        'rxjs': 'rxjs',
+        'rxjs/operators': 'rxjs.operators'
+    },
+    output: {
+        filename: '[name].js',
+        library: 'sequence',     // Name of the package in lower case
+        libraryTarget: 'var',    // Results will be assigned to a variable sequence`
+        path: path.resolve(__dirname, 'dist'),
+    },
+};
+```
+
+Refer to the [webpack documentation] to modify or extend the provided configuration.
```

Comment:
Let's bring back a paragraph that followed after notes on webpack configuration. Perhaps, not here, but in some other place.